### DISCUSSION TITLE
Testing Framework

### DISCUSSION THREAD

**Discussion by gopisaba** [grafana/tanka#1000] [Q&A]
Hi All, 
We are using tanka to generate manifests and flux to deploy them to the cluster. It works nice but we would like to test the Jsonnet and Libsonnet so we don't mess up when we modify them. I came across [jsonnetunit](https://github.com/yugui/jsonnetunit). 

Would like to know what others are using to test their jsonnet/libsonnet?

**Comment by Duologic** [comment]
I authored this one, it is inspired by the one you linked: https://github.com/jsonnet-libs/testonnet

**Reply by gopisaba** [reply]
Thanks @Duologic. will take a look at it.

**Comment by Duologic** [comment]
Additionally to unit testing I've been experimenting with runtime validation techniques here: https://github.com/crdsonnet/validate-libsonnet
